cmd/iq.go

```diff
@@ -13,40 +13,222 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
-
 package cmd
 
 import (
+	"errors"
 	"fmt"
+	"github.com/sirupsen/logrus"
+	"github.com/sonatype-nexus-community/nancy/configuration"
+	"github.com/sonatype-nexus-community/nancy/logger"
+	"os"
+	"path"
 
+	"github.com/mitchellh/go-homedir"
+	"github.com/sonatype-nexus-community/go-sona-types/iq"
+	"github.com/sonatype-nexus-community/nancy/customerrors"
+	"github.com/sonatype-nexus-community/nancy/packages"
+	"github.com/sonatype-nexus-community/nancy/parse"
+	"github.com/sonatype-nexus-community/nancy/types"
 	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+type iqServerFactory interface {
+	create() iq.IServer
+}
+
+type iqFactory struct{}
+
+func (iqFactory) create() iq.IServer {
+	iqServer := iq.New(logLady, iq.Options{
+		User:          viper.GetString(configuration.YamlKeyIQUsername),
+		Token:         viper.GetString(configuration.YamlKeyIQToken),
+		Application:   configIQ.IQApplication,
+		Stage:         configIQ.IQStage,
+		Server:        viper.GetString(configuration.YamlKeyIQServer),
+		OSSIndexUser:  viper.GetString(configuration.YamlKeyUsername),
+		OSSIndexToken: viper.GetString(configuration.YamlKeyToken),
+		Tool:          "nancy-client",
+		DBCacheName:   "nancy-cache",
+		MaxRetries:    300,
+	})
+
+	if logLady != nil && logLady.IsLevelEnabled(logrus.DebugLevel) {
```

Comment:
I'm not sure why you do this check for IsLevelEnabled? Using `.Debug` will limit it to being logged unless you are at `DebugLevel`. Is it just because you don't want to run the cleanUserName etc... stuff?

Reply:
> I'm not sure why you do this check for IsLevelEnabled? Using `.Debug` will limit it to being logged unless you are at `DebugLevel`. Is it just because you don't want to run the cleanUserName etc... stuff?

Oof! Good catch. I think this was leftover thrashing when some unit tests had not yet initialized the null logger. 
Fixed: 4f842ef